File: top150/ArrayString/q135.java
package top150.ArrayString;

public class q135 {
    public int candy(int[] ratings) {
        int n = ratings.length;
        if(n < 2) return n;

        int candy = n;
        int[] front = new int[n];
        int[] back = new int[n];

        for(int i = 1; i < n; i++) {
            if(ratings[i] > ratings[i-1]) front[i] = front[i-1] +1;
        }

        for(int i = n-1; i > 0; i--) {
            if(ratings[i-1] > ratings[i]) back[i-1] = back[i]+1;
        }

        for(int i = 0; i < n; i++) {
            candy += Math.max(front[i], back[i]); 
        }

        return candy;
    }
}

File: top150/ArrayString/q135.py
class Solution:
    def candy(self, ratings):
        n = len(ratings)
        if n <= 1:
            return n
        result = [1] * n
        for i in range(1, n):
            if ratings[i] > ratings[i-1]:
                result[i] = result[i-1] + 1
        for j in range(n-2, -1, -1):
            if ratings[j] > ratings[j+1] and result[j] <= result[j+1]:
                result[j] = result[j+1] + 1
        return sum(result)
